sdk/core/core-util/src/bytesEncoding.browser.ts

```diff
@@ -96,3 +106,11 @@ export function base64UrlToUint8Array(value: string): Uint8Array {
   const base64String = value.replace(/-/g, "+").replace(/_/g, "/");
   return base64ToUint8Array(base64String);
 }
+
+/**
+ * Encodes a hex string into a Uint8Array
+ * @internal
+ */
+export function hexStringToUint8Array(value: string): Uint8Array {
+  return new Uint8Array(value.match(/../g)!.map((x) => parseInt(x, 16)));
```

Comment:
using a regex to split the string into 2-length tuples feels a little like overkill, though I confess I am not sure of an equally fancy one-liner that can do this, since while it's trivial to split on the empty string (or use the string iterator), you'd still have to handle walking it two at a time before doing the parsing.

Out of curiosity, what happens if the string length isn't a multiple of 2? Or if it's empty?

Reply:
you and Jeremy make good points re the use of regex here. I will concede that this one-liner is might be better placed in a code golf tournament than where it is now :)

I will update this to something more traditional.